### ISSUE TITLE
Importing teams with Hebrew names by drag and drop corrupts the text

### ISSUE THREAD

**Issue by alanggreen** [FirstLegoLeagueIL/fllscoring#115]
Drop the attached .csv file into the paste box
The list of teams (under "Data to be loaded") contains the corrupted text, and pressing OK creates teams with corrupted text.
Note that this same team list does work when doing copy and paste from Excel

Other issues
1. "Data to be loaded" has the whole file and not just the teams
2. Not all the data in "Data to be loaded" is actually loaded. This means there is some additional checking not apparent to the user. Confusing
[short brian lee team file with hebrew.zip](https://github.com/FirstLegoLeagueIL/fllscoring/files/1503240/short.brian.lee.team.file.with.hebrew.zip)

![image](https://user-images.githubusercontent.com/6757517/33232429-eebfdbec-d20e-11e7-8b44-80a99412100f.png)

**Comment by taltaub22** [comment]
This happens because the file is not in utf-8 format.
The default encoding in the file Reader api in java script is utf-8.
When I changed the encoding of the file, every thing was working fine.

**Comment by alanggreen** [comment]
How did you change the encoding of the file?
is there an npm package that can detect the file format, and then we can set the FileReader to the format of the file?
In any event, lets leave this till the next version

**Comment by taltaub22** [comment]
Beacuse a csv is essentially a text file, I opened it with notepad++ and encoded it as utf-8.
Right now, everything about the file is happaning client side only.

**Comment by ThinkRedstone** [comment]
https://github.com/aadsm/jschardet/ seems to be able to do it.

**Comment by alanggreen** [comment]
This also looks interesting
https://www.npmjs.com/package/csv-reader
But I like this about jschardet
Stats
91,936 downloads in the last day
559,291 downloads in the last week
4,385,891 downloads in the last month

**Comment by alanggreen** [comment]
> Because a csv is essentially a text file, I opened it with notepad++ and encoded it as utf-8.
> Right now, everything about the file is happening client side only.

Many / most FLL organizers and head refs won't know how to do that.

**Comment by idanstark42** [comment]
You're right. That's a problem. We'll need to look into the program taht generates these files, to understand if the UTF8 comes from there, or if it comes from anywhere else in the process (like sharing it through github)
Anyway, this issue is now about creating a support for non-UTF8 files. @taltaub22 do you want to take this one? You found the cause

**Comment by taltaub22** [comment]
I did not think they know, just saying that means we need to do that process on server side, and not on client side. That means we need to handle file upload and rewrite that feature. I could not find client side library that can do that (beacuse access to file on client side is only available with file reader object). 
Rewrite this feature on server side is our best option.

**Comment by taltaub22** [comment]
@idanstark42. That program is excel.
I can take this. But the simple solution is to rewrite that option with file upload

**Comment by ThinkRedstone** [comment]
Can't you just read the file as bytes, and then use the library on that?
